tests/integration/test_integration.py

```diff
@@ -72,102 +72,100 @@ async def initialize_vault(vault: Vault, timeout: int = 60) -> Tuple[str, str]:
         raise TimeoutError("Timed out waiting for Vault to be ready.")
 
     @staticmethod
-    async def deploy_charm(ops_test, charm: Path) -> None:
+    async def deploy_charm(ops_test: OpsTest, charm: Path) -> None:
         """Deploys charm.
 
         Args:
-            ops_test:
+            ops_test: Ops test Framework.
             charm: Charm path.
-
-        Returns:
-            None
         """
         resources = {
             "vault-image": METADATA["resources"]["vault-image"]["upstream-source"],
         }
-        await ops_test.model.deploy(
+        await ops_test.model.deploy(  # type: ignore[union-attr]
             charm,
             resources=resources,
             application_name=APPLICATION_NAME,
             trust=True,
             series="focal",
         )
 
-    async def post_deployment_tasks(self, namespace: str) -> str:
+    @pytest.mark.abort_on_fail
+    @pytest.fixture(scope="module")
+    async def build_and_deploy(self, ops_test: OpsTest):
+        """Builds and deploys vault-k8s charm.
+
+        Args:
+            ops_test: Ops test Framework.
+        """
+        ops_test.destructive_mode = False
+        charm = await ops_test.build_charm(".")
+        await self.deploy_charm(ops_test, charm)
+
+    @pytest.mark.abort_on_fail
+    async def post_deployment_tasks(self, ops_test: OpsTest) -> str:
         """Runs post deployment tasks as explained in the README.md.
 
         Retrieves Vault's LoadBalancer address, initializes Vault and generates a token for
         the charm.
 
         Args:
-            namespace (str): Kubernetes namespace
+            ops_test: Ops test Framework.
 
         Returns:
             str: Generated token.
         """
-        kubernetes = Kubernetes(namespace=namespace)
+        kubernetes = Kubernetes(namespace=ops_test.model_name)  # type: ignore[arg-type]
         load_balancer_address = await self.wait_for_load_balancer_address(kubernetes=kubernetes)
         vault = Vault(url=f"http://{load_balancer_address}:8200")
-        unseal_key, root_token = await self.initialize_vault(vault=vault)
+        unseal_key, root_token = await self.initialize_vault(vault=vault)  # type: ignore[misc]
         vault.set_token(root_token)
         vault.unseal(unseal_key=unseal_key)
         generated_token = vault.generate_token(ttl="5m")
         return generated_token
 
-    async def test_given_no_config_when_deploy_then_status_is_waiting(  # noqa: E501
-        self, ops_test, charm, cleanup
+    @pytest.mark.abort_on_fail
+    async def test_given_no_config_when_deploy_then_status_is_blocked(  # noqa: E501
+        self, ops_test: OpsTest, build_and_deploy
     ):
-        await self.deploy_charm(ops_test, charm)
-
-        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="blocked", timeout=1000)
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="blocked", timeout=1000)  # type: ignore[union-attr]
```

Comment:
Isn't lint going crazy about the line being too long?

Reply:
No, but I fixed it and it's not complaining.